### PR TITLE
Initial auto-updater

### LINKED ISSUE #21: Keep package list up-to-date
See title.

This likely implies automation work.

### PR DESCRIPTION
**Pull request by JamieMagee**
A first pass at automatically updating package versions. Heavily inspired by https://github.com/vcpkgx/vcpkgx.github.io.

I know that the frontend will need to be updated to deal with this new format, but I wanted to get some feedback before I went too far in on this.

Closes #21

### CODE REVIEW

**Comment by dan-shaw** [comment]
Thanks for the PR! This makes a lot of sense. I am in favor of deprecating the Powershell scripts. I think there are two results missing from `output.json` (the CI baseline results and repository stars), which are used for the package search on the website. Not sure what the best way is to get repository stars from a workflow

Currently `output.json` is not an official API of course, though we will be adding new APIs to the website (for our IDE integration effort). Since we are going to be publishing other APIs, it makes sense to make this API official as well. Let me get back to you on this issue.

**Comment by JamieMagee** [comment]
@dan-shaw Thanks for the feedback. I am a Microsoft FTE, so feel free to reach out to me internally as well (though this effort is designed to help vcpkg support in another open-source project [Renovate](https://github.com/renovatebot/renovate)).

File: scripts/update/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as path from 'path';
+import fs from 'fs-extra';
+import { exit } from 'process';
+import { addVersions } from './tasks/add-versions.js';
+import { readPackages } from './tasks/read-packages.js';
+
+if (!process.argv[2]) {
+  exit(-1);
```
Comment (strega-nil-ms) [inline scripts/update/index.ts:8]: It'd be good to write a message to stderr here saying that this needs an argument (and what that argument is)

**Review by strega-nil-ms — COMMENTED** [review]
The only other thing I see - what about "Stars"?

**Comment by PhoebeHui** [comment]
Related to https://github.com/vcpkg/vcpkg.github.io/issues/48

**Comment by JamieMagee** [comment]
Updated this PR with support for stars. Will follow-up with a PR for CI baseline[^1]

[^1]: https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt

**Comment by JamieMagee** [comment]
Added support for baseline parsing. [Here's a sample `output.json`][1]

[1]: https://gist.github.com/JamieMagee/f043e57b36101b466871e2bef5653406

**Comment by JamieMagee** [comment]
@dan-shaw Is there anyone available to review this?

**Comment by chausner** [comment]
Any status update on this?

**Comment by JavierMatosD** [comment]
Thank you for contributing this pull request to add auto-update functionality. I apologize for the delay in addressing this.

We have since implemented other automation and, due to evolving security policies and plans to move this automation elsewhere, we won't be able to integrate this directly.

We appreciate your contribution and hope you continue sharing your ideas in future PRs.

**Pull request closed without merging**